### PR TITLE
{math}[intel-2016a] Octave 4.0.0 (WIP)

### PR DESCRIPTION
**Pull request by hajgato**

### CODE REVIEW

**Comment by hajgato** [comment]
it fails with some weird error:
https://gist.github.com/hajgato/3da71719f63e049a8e90
relevant codes from header files:
https://gist.github.com/hajgato/f784c2da29fdb64a0c45

**Comment by hajgato** [comment]
bug report submitted http://savannah.gnu.org/bugs/index.php?47373

**Comment by hajgato** [comment]
config.log https://gist.github.com/hajgato/10389cf249832aabb777
config.status https://gist.github.com/hajgato/7f9ff6fa1d1d12bccfe0
config/build output https://gist.github.com/hajgato/bea2eebd8908a86cace2

**Comment by hajgato** [comment]
intel/2016a is affected, 2015b is not. Did the default c-standard changed?

File: easybuild/easyconfigs/o/Octave/Octave-4.0.0-intel-2016a.eb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    ('libGLU', '9.0.0'),
+    ('Qhull', '2015.2'),
+    ('Qt', '4.8.7'),
+    ('HDF5', '1.8.16', '-serial'),
+    ('qrupdate', '1.1.2'),
+    ('SuiteSparse', '4.5.1', '-METIS-5.1.0'),
+    ('GraphicsMagick', '1.3.23'),
+]
+
+preconfigopts = 'LLVM_CONFIG = $EBROOTLLVM/bin/llvm-config '
```
Comment (boegel) [inline easybuild/easyconfigs/o/Octave/Octave-4.0.0-intel-2016a.eb]: missing `&&` at the end

Comment (hajgato) [inline easybuild/easyconfigs/o/Octave/Octave-4.0.0-intel-2016a.eb]:
```
configure: WARNING: llvm/IR/Verifier.h: present but cannot be compiled
configure: WARNING: llvm/IR/Verifier.h:     check for missing prerequisite headers?
configure: WARNING: llvm/IR/Verifier.h: see the Autoconf documentation
configure: WARNING: llvm/IR/Verifier.h:     section "Present But Cannot Be Compiled"
configure: WARNING: llvm/IR/Verifier.h: proceeding with the compiler's result
configure: WARNING:     ## ------------------------------------------ ##
configure: WARNING:     ## Report this to http://octave.org/bugs.html ##
configure: WARNING:     ## ------------------------------------------ ##
```
So I deleted Just-In-Time compilation and the `LLVM` dep.

**Comment by hajgato** [comment]
bug report at `GNULIB`: http://lists.gnu.org/archive/html/bug-gnulib/2016-03/msg00026.html
bug report at Intel: https://software.intel.com/en-us/forums/intel-c-compiler/topic/612811

**Comment by boegel** [comment]
@hajgato Thanks for following up on this... Based on the fact that this works just fine with older versions of the Intel compilers, it does look like a bug in the 2016 version of the Intel compilers.

**Comment by hajgato** [comment]
It has been confirmed that this is a compiler bug. There is no workaround, it will be corrected in the next release.

**Comment by boegel** [comment]
this can be closed, superseded by #3717?

**Pull request closed without merging**